File: Math/euclidean.py
# 유클리드 호제법
def euclidean(a, b):
    while b != 0:
        a, b = b, a % b
    return a


# Extended Euclidean Algorithm, 확장된 유클리드 알고리즘
def EEA(a, b):
    if a == 0:
        return b, 0, 1

    gcd, x1, y1 = EEA(b % a, a)

    x = y1 - (b // a) * x1
    y = x1

    return gcd, x, y

File: Math/euclidean.cpp
#include <algorithm>
#include <iostream>

using namespace std;

int euclidean(int X, int Y) {
  if (X < Y)
    swap(X, Y);

  while (Y != 0) {
    int Temp = X % Y;
    X = Y;
    Y = Temp;
  }
  return X;
}

int main() {
  int X, Y;
  cin >> X >> Y;
  cout << euclidean(X, Y);
}